autorag/nodes/passagereranker/colbert.py

```diff
@@ -55,22 +58,22 @@ def colbert_reranker(queries: List[str], contents_list: List[List[str]],
 
 
 def colbert_run_model(contents_list, model, tokenizer, device, batch_size: int):
-    batch_contents_list = make_batch(contents_list, batch_size)
+    flattened_queries, flattened_contents = map(list, zip(*contents_list))
+    batch_queries = make_batch(flattened_queries, batch_size)
+    batch_contents = make_batch(flattened_contents, batch_size)
     results = []
 
-    for batch_contents in batch_contents_list:
-        flattened_batch_queries, flattened_batch_contents = map(list, zip(*batch_contents))
-
+    for batch_queries, batch_contents in zip(batch_queries, batch_contents):
         # Tokenize both queries and contents together
-        feature = tokenizer(flattened_batch_queries, flattened_batch_contents, padding=True, truncation=True,
+        feature = tokenizer(batch_queries, batch_contents, padding=True, truncation=True,
                             return_tensors="pt").to(device)
 
         # Process the combined feature through the model in one go
         outputs = model(**feature)
         last_hidden_state = outputs.last_hidden_state
 
         # Split the embeddings back into query and content parts
-        num_queries = len(flattened_batch_queries)
+        num_queries = len(batch_queries)
         query_embedding, content_embedding = last_hidden_state.split(
             [num_queries, last_hidden_state.size(1) - num_queries], dim=1)
```

Comment:
It looks like the real problem is not `batch` in here. The parallel code itself looks wrong. 

Look at #326.

You can see the original code, and the new code that you pushed.

The original code is like below.

```python
def get_colbert_score(query: str, content_list: List[str],
                      model, tokenizer) -> List[float]:
    query_encoding = tokenizer(query, return_tensors="pt")
    query_embedding = model(**query_encoding).last_hidden_state
    rerank_score_list = []

    for document_text in content_list:
        document_encoding = tokenizer(
            document_text, return_tensors="pt", truncation=True, max_length=512
        )
        document_embedding = model(**document_encoding).last_hidden_state

        sim_matrix = torch.nn.functional.cosine_similarity(
            query_embedding.unsqueeze(2), document_embedding.unsqueeze(1), dim=-1
        )

        # Take the maximum similarity for each query token (across all document tokens)
        # sim_matrix shape: [batch_size, query_length, doc_length]
        max_sim_scores, _ = torch.max(sim_matrix, dim=2)
        rerank_score_list.append(torch.mean(max_sim_scores, dim=1))

    return list(map(float, rerank_score_list))
```

The colbert model acts like 'embedding_model' in here. The `last_hidden_state` is embedding vector. 
It gets cosine similarity between each query_embedding and each content_embedding. And rerank contents by that score.

Return to your code. It is much easier you look at #326

The first thing you did wrong is tokenize query and contents at once. 

```python
# Tokenize both queries and contents together
        feature = tokenizer(flattened_batch_queries, flattened_batch_contents, padding=True, truncation=True,
                            return_tensors="pt").to(device)
```
If you decode the `feature` you can get the result like below.

```
'[CLS] what is the capital of france? [SEP] nomadamas is great team [SEP]'
```
It means, it encodes the query and content in the one sentence. And you put this feature into model 'at once'. 
I think you try to split the result of embedding at the next code, but it does not work that way at all.
Model creates embedding of the tokenized sentence. It did not create embedding each sentence separately, it embeds a whole input string. So splitting it after embedding seems useless.

In conclusion, the code itself after #326 is wrong and might get the bad results and occurs several errors.